docs/marketplace/tutorials/fully-functioning-rag-application.md

```diff
@@ -0,0 +1,172 @@
+
+- `MONGODB_CLUSTER_URI`: This is the full connection string to MongoDB.
+- `EMBEDDING_API_KEY`: An API key to use when communicating with OpenAI for generating services. It should be the same as the OpenAI API key.
+- `LLM_API_KEY`: An API key to use when communicating with OpenAI for generating services. It should also be the same as the OpenAI API key.
+
+:::note
+`EMBEDDING_API_KEY` and `LLM_API_KEY` will have the same value: they are the API key that you can generate from [OpenAI](https://platform.openai.com).
+:::
+
+:::warning
+Since this is information that you don't want to have visible to anyone, it is preferable to include this information as [project variables](/console/project-configuration/manage-environment-variables/index.md).
+:::
+
+After updating the environment variables of the service, it is time to update the config map: click on the _ConfigMaps & Secrets_ tab and you will be redirected to the Config Map configuration page.
+
+The _AI RAG Template_ is created with a precompiled config map that includes all the required keys, but the values need to be included.
+
+To be more specific, the configuration should be as follows:
+
+```json
+{
+  "llm": {
+    "name": "gpt-4o-mini"
+  },
+  "embeddings": {
+    "name": "text-embedding-3-small"
+  },
+  "vectorStore": {
+    "dbName": "rag-database",
+    "collectionName": "rag-collection",
+    "indexName": "vector_index",
+    "relevanceScoreFn": "cosine",
+    "embeddingKey": "embedding",
+    "textKey": "text",
+    "maxDocumentsToRetrieve": 3
+  }
+}
+```
+
+To be more specific:
+
+- the `llm.name` key is the LLM model among those available from [OpenAI](https://platform.openai.com/docs/models), suggested are `gpt-4o` and `gpt-4o-mini`
+- the `embeddings.name` key is the Embedding model used to generate embeddings, among those available from [OpenAI](https://platform.openai.com/docs/models/embeddings), the suggested is `text-embedding-3-small`
+- the `vectorStore.dbName` is the name of the database where the embeddings will be saved
+- the `vectorStore.collectionName` is the name of the collection where the embeddings will be saved
+- the `vectorStore.indexName` is the name of the [MongoDB Search Vector Index](https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-overview/), with `vector_index` as suggested name; this is a particular index that the service will automatically create or update at the startup
+- the `vectorStore.relevanceScoreFn` is the name of the _similarity search function_ used to retrieve the embedding documents; MongoDB includes three different available functions: `cosine` (suggested), `euclidean` and `dotProduct`
+- the `vectorStore.embeddingKey` is the name of the field where to save the embedding of a single document, in the shape of a multidimensional array; we suggest using the default value `embedding`
```

Comment:
```suggestion
- the `vectorStore.embeddingKey` is the name of the field where the embedding of a single document is saved, in the shape of a multidimensional array; we suggest using the default value `embedding`
```

Reply:
Um, actually `embedding` should be plural..

```suggestion
- the `vectorStore.embeddingKey` is the name of the field where the embeddings of a single document are saved, in the shape of a multidimensional array; we suggest using the default value `embedding`
```